pkg/services/ngalert/notifier/config.go

```diff
@@ -0,0 +1,77 @@
+package notifier
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/grafana/alerting-api/pkg/api"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+func PersistTemplates(cfg *api.PostableUserConfig, path string) ([]string, bool, error) {
+	if len(cfg.TemplateFiles) < 1 {
+		return nil, false, nil
+	}
+
+	var templatesChanged bool
+	paths := make([]string, 0, len(cfg.TemplateFiles))
+	for name, content := range cfg.TemplateFiles {
+		if name != filepath.Base(filepath.Clean(name)) {
+			return nil, false, fmt.Errorf("template file name '%s' is  not valid", name)
+		}
+
+		err := os.MkdirAll(path, 0750)
+		if err != nil {
+			return nil, false, fmt.Errorf("unable to create template directory %q: %s", path, err)
+		}
+
+		file := filepath.Join(path, name)
+		paths = append(paths, file)
+
+		// Check if the template file already exists and if it has changed
+		// We can safeily ignore gosec here and we've previously checked the filename is clean
+		// nolint:gosec
+		if tmpl, err := ioutil.ReadFile(file); err == nil && string(tmpl) == content {
+			// Templates file is the same we have, no-op and continue.
+			continue
+		} else if err != nil && !os.IsNotExist(err) {
+			return nil, false, err
+		}
+
+		if err := ioutil.WriteFile(file, []byte(content), 0644); err != nil {
+			return nil, false, fmt.Errorf("unable to create Alertmanager template file %q: %s", file, err)
+		}
+		// nolint:gosec
+
+		templatesChanged = true
+	}
+
+	return paths, templatesChanged, nil
+}
+
+func Load(rawConfig string) (*api.PostableUserConfig, error) {
+	cfg := &api.PostableUserConfig{}
+
+	if err := yaml.UnmarshalStrict([]byte(rawConfig), cfg); err != nil {
+		return nil, errors.Wrap(err, "Unable to parse Alertmanager configuration")
```

Comment:
Go errors shouldn't be capitalized.
```suggestion
		return nil, errors.Wrap(err, "unable to parse Alertmanager configuration")
```